### ISSUE TITLE
Unable to recreate workspace after deletion

### ISSUE THREAD

**Issue by jderieg** [AzBuilder/terrakube#522]
### Bug description 🐞

If I delete a workspace, I can't create a new workspace with the same name. It seems that it doesn't actually get deleted from the database since I get a duplicate key error in the API.

```
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "const_workspace"
  Detail: Key (organization_id, name)=(c0842cc6-3f3b-401f-8f49-e9a5859ca282, testworkspace) already exists.
```

### Steps to reproduce

- Create workspace
- Delete workspace
- Attempt to create workspace with same name (the UI doesn't give any error at all, just doesn't respond). The API throws an error about duplicate key.

### Expected behavior

Should be able to create a new workspace with the same name as one that existed previously but has been deleted.

**Comment by alfespa17** [comment]
Hello @jderieg 

This is happening because we don't delete it from the database, we change the "flag" to "deleted=true"

https://github.com/AzBuilder/terrakube/blob/main/ui/src/domain/Workspaces/Details.jsx#L202

We only show workspaces where the flag is "false" in this database entity.
https://github.com/AzBuilder/terrakube/blob/main/api/src/main/java/org/terrakube/api/rs/workspace/Workspace.java#L33

This could be an improvement but it will require some work because this part of the code is using [Elide](https://github.com/yahoo/elide), I will try to add it for the next version, for now a workaround could be to use some "suffix" in the name.

**Comment by alfespa17** [comment]
Partially fixed by #526, It will allow to create a workspace with the same name but it it won't delete the data from the backend storage
